src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -555,6 +570,75 @@ func (h *PromWriteHandler) forward(
 	return nil
 }
 
+func (h *PromWriteHandler) maybeLogSeriesWithLongestLabels(logger *zap.Logger, timeSeries []prompb.TimeSeries) {
+	if len(timeSeries) == 0 {
+		return
+	}
+
+	n := atomic.LoadInt32(&h.numLiteralIsTooLong)
+	if n >= maxLiteralIsTooLongLogCount || !atomic.CompareAndSwapInt32(&h.numLiteralIsTooLong, n, n+1) {
```

Comment:
nit: I was also thinking maybe we could log once a minute for example. I guess implementation would be similar - constant for "how often" and "last logged time". This would allow to get pretty recent errors. And potentially catch multiple problems that might be causing large labels.